include/boost/math/fft/algorithms.hpp

```diff
@@ -489,8 +489,59 @@
     }
   }
   
+  template <class T>
+  void complex_dft_power2(const T *in_first, const T *in_last, T* out, int sign)
+  {
+    /*
+      Cooley-Tukey mapping, in-place Decimation in Time 
+    */
+    const long ptrdiff = static_cast<long>(std::distance(in_first,in_last));
+    if(ptrdiff <=0 )
+      return;
+    const long n = lower_bound_power2(ptrdiff);
+    
+    if(in_first!=out)
+      std::copy(in_first,in_last,out);
+    
+    if (n == 1)
+        return;
+
+
+    // Gold-Rader bit-reversal algorithm.
+    for(int i=0,j=0;i<n-1;++i)
+    { 
+      if(i<j)
+        std::swap(out[i],out[j]);
+      for(int k=n>>1;!( (j^=k)&k );k>>=1);
+    }
+    
+    
+    for (int len = 2, k = 1; len <= n; len <<= 1, ++k)
+    {
+      for (int i = 0; i < n; i += len)
+      {
+        {
+          int j=0;
+          T* u = out + i + j, *v = out + i + j + len / 2;
+          T Bu = *u, Bv = *v;
+          *u = Bu + Bv;
+          *v = Bu - Bv;
+        }
+        for (int j = 1; j < len / 2; ++j)
+        {
+          T cs{complex_root_of_unity<T>(1<<k,j*sign)};
+          
+          T* u = out + i + j, *v = out + i + j + len / 2;
+          T Bu = *u, Bv = *v * cs;
+          *u = Bu + Bv;
+          *v = Bu - Bv;
+        }
+      }
+    }
+  }
+  
   template<class complex_value_type>
-  void complex_dft_power2(
+  void complex_dft_power2_dif(
```

Comment:
According to the GSL documentation, the rfft can be deduced from the decimation-in-time (DIT) complex fft.
The default complex fft for powers of 2 is implemented as Decimation-in-frequency (DIF), i wrote another complex fft as DIT to test the algorithm before writing the rfft.